python/opening_hours.pyi

```diff
@@ -0,0 +1,152 @@
+from datetime import datetime
+from typing import Self
+
+def validate(oh: str) -> bool:
+    """
+    Validate that input string is a correct opening hours description.
+
+    Examples
+    --------
+    >>> opening_hours.validate("24/7")
+    True
+    >>> opening_hours.validate("24/24")
+    False
+    """
+
+class RangeIterator:
+    def __iter__(self) -> Self:
+        ...
+
+    def __next__(self) -> tuple[datetime, datetime, str, list]:
+        ...
+
+class OpeningHours:
+    """
+    Class for parsing input opening hours description.
+
+    Raises
+    ------
+    SyntaxError
+        Given string is not in valid opening hours format.
+
+    Examples
+    --------
+    >>> oh = OpeningHours("24/7")
+    >>> oh.is_open()
+    True
+    """
```

Comment:
Is it recommended to put docstrings in this file?

It is redundant with those in *lib.rs*  and it appears the ones in *opening_hours.py* are not added to `__doc__`, so maybe we can remove them from here?

Reply:
I think that coding tools are able to use these docstrings and present them to the user. For example, using a pyright LSP server in neovim, I can get hover documentation. Without the docstrings, it looks like this:

![Screenshot 2024-11-18 at 10 35 59](https://github.com/user-attachments/assets/d9198df7-b8ff-46c8-95ba-d52fee5012b4)

With the docstrings, I get:

![Screenshot 2024-11-18 at 10 36 11](https://github.com/user-attachments/assets/9e41438d-eb1e-4f61-a4ed-aa03bf6e6d1c)